src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+  var num_illumina_long_switch_sites: Long = 0,
+  var frac_phased: Double = 0,
+  var frac_phased_with_truth_phased: Double = 0,
+  var frac_truth_phased_in_called_block: Double = 0,
+  var frac_phased_with_truth_phased_in_called_block: Double = 0,
+  var mean_called_block_length: Double = 0,
+  var median_called_block_length: Double = 0,
+  var stddev_called_block_length: Double = 0,
+  var n50_called_block_length: Double = 0,
+  var n90_called_block_length: Double = 0,
+  var l50_called_block_length: Double = 0,
+  var mean_truth_block_length: Double = 0,
+  var median_truth_block_length: Double = 0,
+  var stddev_truth_block_length: Double = 0,
+  var n50_truth_block_length: Double = 0,
+  var n90_truth_block_length: Double = 0,
+  var l50_truth_block_length: Double = 0
+) extends Metric {
+  private def divide(a: Double, b: Double): Double = {
+    if (b == 0) 0
+    else a / b
+  }
+  def finalizeMetric(): this.type = {
+    this.frac_phased                                   = this.divide(this.num_phased, this.num_called)
+    this.frac_phased_with_truth_phased                 = this.divide(this.num_phased_with_truth_phased, this.num_called_with_truth_phased)
+    this.frac_truth_phased_in_called_block             = this.divide(this.num_truth_phased_in_called_block, this.num_truth_phased)
+    this.frac_phased_with_truth_phased_in_called_block = this.divide(this.num_both_phased_in_called_block, this.num_truth_phased_in_called_block)
+    this
+  }
+}
+
+/** The cigar Elements */
+private object CigarType extends Enumeration {
+  type CigarType = Value
+  val Match,      // 0
+  Mismatch,       // 1
+  Truth,          // 2
+  Call,           // 3
+  TruthEnd,       // 4
+  CallEnd,        // 5
+  BothEnd = Value // 6
+}
+
+private[vcf] object Cigar {
+  import CigarType._
+
+  type Cigar = Seq[CigarType]
+
+  private type VCtx = VariantContext
+
+  /** Creates a Cigar from an iterator over variant contexts from a truth and call sample.
+    *
+    * If `skipMismatchingAlleles`, then it skips sites where both truth and call have a variant call, but the alleles
+    * disagree between the two.
+    *
+    * Examples of short and long errors:
+    *   CALL :  MMMMXXXMMM   MMMM|FFFFFFF|MMMMMFMMMM|F|MMFFFMMM
+    *   TRUTH:  MMMM   MMMMMMMMMM-MMMMMMM|MMMMMMMMMM|F|MMMMMMMM
+    *             correct       long        short     long
+    *   CIGAR:  00003330002220000511111116000001000060600111000
+    * shows one known block split
+    */
+  def toCigar(pairedIterator: JointVariantContextIterator,
+              truthPhaseBlockDetector: OverlapDetectorByContig,
+              callPhaseBlockDetector: OverlapDetectorByContig,
+              metric: AssessPhasingMetric,
+              skipMismatchingAlleles: Boolean): Cigar = {
+    val cigar: Cigar = pairedIterator
+      .filter { // ensure the alleles are the same when both truth and call are called
+        case Seq(Some(t: VCtx), Some(c: VCtx)) =>
```

Comment:
`JointVariantContextIterator` can handle more than two samples!  I can map it to a tuple, then the parameter's signature would be `Iterator[Tuple2[Option[VariantContext], Option[VariantContext]]]`